### DISCUSSION TITLE
Built with Nix

### DISCUSSION THREAD

**Discussion by t-gebauer** [Interlisp/medley#169] [Show and tell]
I created a simple nix expression to run Medley on my [NixOS](https://nixos.org) system.
Can build this on any system with the Nix package manager installed.
The Maiko build process is straight-forward, but maybe this saves someone a few key strokes ;)

Can be build with `nix-build medley.nix`

medley.nix
```nix
with import <nixpkgs> { };
let
  maiko = stdenv.mkDerivation {
    pname = "maiko";
    version = "git-2021-01-27";
    src = fetchgit {
      url = "https://github.com/Interlisp/maiko";
      rev = "405b55509e5c6e103e0943516d40eb90977e7099";
      sha256 = "19n34z0wf1k1cdb8kbcvld0i42qhf0ajhmr72i2z0r71h42psaxr";
    };
    buildInputs = [clang gnumake xorg.libX11];
    buildPhase = ''
      cd bin
      ./makeright x
      cd ..
    '';
    installPhase = ''
      mkdir $out
      cp -r bin $out
      # Export all generated directories
      # (nix sets the mtime of all fetched files to 1970-01-01)
      find . -maxdepth 1 -mtime -1 -exec cp -r {} $out \;
    '';
  };
  medley = fetchgit {
    url = "https://github.com/Interlisp/medley";
    rev = "f2d3bafdf92ff55e0fad70bbca710125f062ce5f";
    sha256 = "00x0bfvagjijl3khghxq53acch3lkx0xqnpjiilx96p1rb0whcnw";
  };
in
writeShellScript "medley" ''
  env MAIKODIR=${maiko} ${medley}/run-medley "$@"
''
```

PS: My nix-fu is not very strong, so there are probably better ways to do this.

**Comment by nbriggs** [comment]
Nice.  I don't know anything about NixOS/nix package manager but I suspect we can incorporate this as a starting point for someone who might have suggestions.   We've been turning the crank on maiko emulator builds pretty frequently but it looks easy enough to update the script when we reach stable points.

**Comment by waywardmonkeys** [comment]
I've submitted Interlisp/maiko#311 and Interlisp/maiko#312 that would remove the need for your two `substituteInPlace` invocations.

But you might also find the `cmake` build easy to use as well and that might be easier to make cross-platform, especially if I add an `install` target there.

**Reply by t-gebauer** [reply]
Thank you for the PRs. Removed `substituteInPlace`.
